akka-http-tests/src/test/scala/akka/http/scaladsl/server/DiscardEntityDefaultRejectionHandlerSpec.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.server
+
+import akka.http.scaladsl.model.ContentTypes.`text/plain(UTF-8)`
+import akka.http.scaladsl.model.HttpEntity
+import akka.http.scaladsl.model.StatusCodes.NotFound
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import org.scalatest.concurrent.Eventually._
+import org.scalatest.concurrent.ScalaFutures
+
+class DiscardEntityDefaultRejectionHandlerSpec extends RoutingSpec with ScalaFutures {
+
+  private val route = path("foo") {
+    complete("bar")
+  }
+
+  private var elementsEmitted = 0
```

Comment:
Consumed from a stream so needs to be `@volatile` (I think that is enough since there will never be any actual concurrent access)

Reply:
Right!